include/maxscale/dcb.h

```diff
@@ -176,11 +178,13 @@ typedef struct dcb
     bool            ssl_write_want_read;    /*< Flag */
     bool            ssl_write_want_write;    /*< Flag */
     bool            was_persistent;  /**< Whether this DCB was in the persistent pool */
+    bool            high_water_has_reached; /** High water mark has reached, to determine whether need release throttle */
     struct
     {
         struct dcb *next; /**< Next DCB in owning thread's list */
         struct dcb *tail; /**< Last DCB in owning thread's list */
     } thread;
+    struct dcb *next_backend;        /** Next Backend DCB in owning session's list */
```

Comment:
This should not be here. A DCB is basically an abstraction for a connection, while the session object binds together one client connection and indirectly a number of backend connections. With this change, that separation of concerns is broken and that is not good.